x-pack/plugins/siem/public/common/components/exceptions/helpers.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiText, EuiCommentProps, EuiAvatar } from '@elastic/eui';
+import { flatten, capitalize } from 'lodash';
+import moment from 'moment';
+
+import * as i18n from './translations';
+import {
+  FormattedEntry,
+  OperatorType,
+  OperatorOption,
+  ExceptionEntry,
+  NestedExceptionEntry,
+  DescriptionListItem,
+  Comment,
+  ExceptionListItemSchema,
+} from './types';
+import { EXCEPTION_OPERATORS, isOperator } from './operators';
+
+export const getEmptyValue = () => '—';
+
+/**
+ * Returns the operator type, may not need this if using io-ts types
+ *
+ * @param entry a single ExceptionItem entry
+ */
+export const getOperatorType = (entry: ExceptionEntry): OperatorType => {
+  switch (entry.type) {
+    case 'nested':
+    case 'match':
+      return OperatorType.PHRASE;
+    case 'match_any':
+      return OperatorType.PHRASES;
+    case 'list':
+      return OperatorType.LIST;
+    default:
+      return OperatorType.EXISTS;
+  }
+};
+
+/**
+ * Determines operator selection (is/is not/is one of, etc.)
+ * Default operator is "is"
+ *
+ * @param entry a single ExceptionItem entry
+ */
+export const getExceptionOperatorSelect = (entry: ExceptionEntry): OperatorOption => {
+  const operatorType = getOperatorType(entry);
+  const foundOperator = EXCEPTION_OPERATORS.find((operatorOption) => {
+    return entry.operator === operatorOption.operator && operatorType === operatorOption.type;
+  });
+
+  return foundOperator ?? isOperator;
+};
+
+export const determineIfIsNested = (
+  tbd: ExceptionEntry | NestedExceptionEntry
+): tbd is NestedExceptionEntry => {
+  if (tbd.type === 'nested') {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Formats ExceptionItem entries into simple field, operator, value
+ * for use in rendering items in table
+ *
+ * @param entries an ExceptionItem's entries
+ */
+export const getFormattedEntries = (
+  entries: Array<ExceptionEntry | NestedExceptionEntry>
+): FormattedEntry[] => {
+  const formattedEntries = entries.map((entry) => {
+    if (determineIfIsNested(entry)) {
+      const parent = { fieldName: entry.field, operator: null, value: null, isNested: false };
+      return entry.entries.reduce<FormattedEntry[]>(
+        (acc, nestedEntry) => {
+          const formattedEntry = formatEntry({
+            isNested: true,
+            parent: entry.field,
+            item: nestedEntry,
+          });
+          return [...acc, { ...formattedEntry }];
+        },
+        [parent]
+      );
+    } else {
+      return formatEntry({ isNested: false, item: entry });
+    }
+  });
+
+  return flatten(formattedEntries);
```

Comment:
I think instead of lodash we can just use flatten directly off of the array here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

```ts
return formattedEntries.flat();
```

That would remove another lodash'ism that native JS provides.

Reply:
Good call. Updated - thanks!